transit/script/screen.py

```diff
@@ -0,0 +1,123 @@
+    IB_sum["Screenline"] = Screenline
+    IB_sum["Direction"] = "IB"
+    IB_sum["Operator"] = Operator
+    IB_sum["Mode"] = Mode
+
+    # Filter rows for OB and calculate the sum of 'AB_VOL'
+    ST_TOD_OB = ST_TOD[(ST_TOD["A"].isin(B)) & (ST_TOD["B"].isin(A))]
+    OB_sum = ST_TOD_OB.groupby("TOD")["AB_VOL"].sum().reset_index()
+    OB_sum["Screenline"] = Screenline
+    OB_sum["Direction"] = "OB"
+    OB_sum["Operator"] = Operator
+    OB_sum["Mode"] = Mode
+
+    ST_TOD = pd.concat([IB_sum, OB_sum])
+    ST_TOD = ST_TOD.rename(columns={"AB_VOL": "Ridership"})
+
+    return ST_TOD
+
+
+def screen_df(file_name, HWY_SCREENS):
+    screenline_total = []
+    for i in HWY_SCREENS.keys():
+        screenline = process_data(
+            file_name,
+            HWY_SCREENS[i][2][0],
+            HWY_SCREENS[i][0],
+            HWY_SCREENS[i][1],
+            HWY_SCREENS[i][2][1],
+            HWY_SCREENS[i][2][2],
+            HWY_SCREENS[i][2][3],
+        )
+        screenline["Key"] = (
+            screenline["Screenline"]
+            + screenline["Operator"]
+            + screenline["TOD"]
+            + screenline["Direction"]
+        )
+        screenline = screenline[
+            ["Screenline", "Direction", "TOD", "Key", "Ridership", "Operator", "Mode"]
+        ]
+        screenline = screenline.sort_values(by="Direction").reset_index(drop=True)
+        screenline_total.append(screenline)
+    model_Screenlines = pd.concat(screenline_total)
+    return model_Screenlines
+
+
+def concat_final_SL(file_name, output_transit_dir, model_BART_SL, model_SL):
+    model_Screenlines = screen_df(file_name, HWY_SCREENS)
+    BART_Screenlines = pd.read_csv(output_transit_dir / model_BART_SL)
+    BART_Screenlines["Operator"] = "BART"
+    BART_Screenlines["Mode"] = "BART"
+    BART_Screenlines["Key"] = (
+        BART_Screenlines["Screenline"]
+        + BART_Screenlines["Operator"]
+        + BART_Screenlines["TOD"]
+        + BART_Screenlines["Direction"]
+    )
+    model_SL_df = pd.concat([BART_Screenlines, model_Screenlines]).reset_index(
+        drop=True
+    )
+    model_SL_df.to_csv(output_transit_dir / model_SL, index=False)
+
+
+if __name__ == "__main__":
+    with open("transit.toml", "rb") as f:
+        config = tomllib.load(f)
+
+    transit_line_rename_filepath = (
+        config["directories"]["resources"] / config["transit"]["line_rename_filename"]
+    )
+    transit_validation_2019_alfaro_filepath = config["transit"][
+        "transit_validation_2019_alfaro_filepath"
+    ]
+    model_run_dir = config["directories"]["model_run"]
+    model_BART_SL = config["output"]["model_BART_SL"]
+    model_SL = config["output"]["model_SL"]
+    output_dir = model_run_dir / "validation_workbook" / "output"
+    output_transit_dir = output_dir / "transit"
+    output_transit_dir.mkdir(parents=True, exist_ok=True)
+    time_periods = ["EA", "AM", "MD", "PM", "EV"]
+    dbf_file = read_transit_assignments(model_run_dir, time_periods)
```

Comment:
we should name this variable by what's in the file, rather than by the original file format